app/components/CardBet.tsx

```diff
@@ -1,39 +1,74 @@
 "use client";
 
 import { Card } from "@/app/components/ui";
-import { FixedProductMarketMaker } from "@/queries/omen";
+import { UserPosition } from "@/queries/conditional-tokens/types";
+import { FixedProductMarketMaker, getConditionMarket } from "@/queries/omen";
 import { remainingTime } from "@/utils/dates";
+import { useQuery } from "@tanstack/react-query";
 import { cx } from "class-variance-authority";
 import Link from "next/link";
 import { Button, Logo, Tag } from "swapr-ui";
+import { fromHex } from "viem";
 
-interface CardBetProps {
-  market: FixedProductMarketMaker;
+interface BetProps {
+  userPosition: UserPosition;
 }
 
-export const CardBet = ({ market }: CardBetProps) => {
+export const CardBet = ({ userPosition }: BetProps) => {
+  const conditionId = userPosition.position.conditionIdsStr;
+
+  const { data, isLoading, isFetching } = useQuery({
+    queryKey: ["getConditionMarket", conditionId],
+    queryFn: async () =>
+      getConditionMarket({
+        id: conditionId,
+      }),
+    enabled: !!conditionId,
+  });
+
+  if (isLoading || isFetching) return <LoadingCardBet />;
+
+  const market: FixedProductMarketMaker = data?.conditions[0]
+    ?.fixedProductMarketMakers[0] as FixedProductMarketMaker;
+
   const closingDate = new Date(+market.openingTimestamp * 1000);
+  const answer =
+    market.question.currentAnswer &&
+    fromHex(market.question.currentAnswer, "number");
+  const outcomeIndex = userPosition.position.indexSets[0];
 
-  const outcomes = ["win", "loss", "-"];
-  const randomOutcome = outcomes[Math.floor(Math.random() * outcomes.length)];
+  const outcomes = userPosition.position.conditions[0].outcomes;
+  const outcomeString = outcomes ? outcomes[outcomeIndex - 1] : "";
+
+  const hasFinished = !!answer;
```

Comment:
We can use this instead of answer in line 47 and 116

Reply:
I think in that case is more clear what answer == outcomeIndex then hasFinished == outcomeIndex